src/pynwb/file.py

```diff
@@ -55,7 +56,8 @@ class Subject(NWBContainer):
             {'name': 'subject_id', 'type': str, 'doc': 'a unique identifier for the subject', 'default': None},
             {'name': 'weight', 'type': str, 'doc': 'the weight of the subject', 'default': None},
```

Comment:
Can we add the unit for 'weight' or describe how the string should be formatd. .

Seems like a candidate for changing from str to float and then formatting the string in the API.

Reply:
I updated the docstring and now allow float in the constructor with a recommendation to use that.

I am wary of allowing only float. That would be a good change for a major release.

It would be nice to change this in the schema to be float, with a 'unit' attribute saying that it is in kg.

Reply:
I agree. Allowing both float and str makes sense. Updating it in the schema makes sense as well. I'll create an issue for this. There are a few other things that are problematic with this field.